test/common/src/main/kotlin/org/codice/compliance/verification/binding/RedirectBindingVerifier.kt

```diff
@@ -178,10 +204,7 @@ class RedirectBindingVerifier(private val response: IdpRedirectResponseDecorator
         val encodedRelayState = response.relayState
         val givenRelayState = response.isRelayStateGiven
 
-        if (encodedRelayState == null) {
-            if (givenRelayState) {
-                throw SAMLComplianceException.create(SAMLBindings_3_4_3_b1, message = "RelayState not found.")
-            }
+        if (!givenRelayState && encodedRelayState == null) {
```

Comment:
✏️ Same comment about checking the relay state before actually calling the method.